languages/python/reference/exercise-concepts/matrix.md

```diff
@@ -0,0 +1,39 @@
+
+[Alternate Implementation](https://exercism.io/tracks/python/exercises/matrix/solutions/e5004e990ddc4582a50ecc1f660c31df)
+
+[Extended Implmentation](https://exercism.io/tracks/python/exercises/matrix/solutions/b6a3486a35c14372b64fdc35e7c6f98f)
+
+
+ ## General  
+
+  - classes:  the exercise objective is to define a `matrix` type.  Tested methods are linked to a `matrix` class
+  - objects:  creating different instances with different data representing different `matrices` is tested
+  - constructor:  customizing object initalization with actions and persisting data.  The example uses a constructor to process the passed in data into a list of lists assigned to an instance property
+  - magic methods: the example uses the `__init__` magic method as its constructor for the class 
+  - return values: "row" and "column" list values are expected from defined instance method(s)
+  - `self`:  the example uses this keyword for methods and properties linked to a specific instance of the class  
+  - namespaces:  knowing to use `self`.`<propertyname>` for instance properties and `self` as first argument to instance methods in a class
+  - instance methods: tests for this exercises require one or more instance methods that will return a specified row or column list of the `matrix`. 
+  - instance properties:  this exercise rquires one or more instance properties to persist passed in data.
+  - mutability: in the extended example, knowing there are no protected or private properties in python and adjusting coding patterns
+  - assignment:  instance properties need to be assigned passed in data
+  - method arguments:  the methods returning "row" and "column" need to take both `self` and an integer as arguments
+  - lists:  this exercise requires "row" or "column" be returnd as a `list`. A `list` of `lists` is also the reccommended way to process and store the passed-in data.
+  - indexing: the "rows" and "columns" of this exercise need to be retrieved from a list of lists via index 
+  - bracket notation: knowing that `[]` should be used to refer to a value at a specific index in a list
+  - slicing:  the extended solution to this exercise can employ a slice (returns a copy) instead of calling `.copy()`.
+  - iteration:  the passed-in string is iterated over, and split into rows.  The row lists are iterated over and split into elements
+  - iterables:  understanding that strings, lists, and other data structures can be iterated over in the same fashion
+  - iterators:  the example solution for this exercise uses `zip()`, which returns an _iterator_.
+  - for loop: iterating over the passed in `matrix` string using a `for` loop to extract "rows" and "columns" that are appended to a list
+  - comprehension syntax:  knowing that this is equivelent to a `for loop` - putting the row or column creation code _inside_ the list literal instead of using loop + append.
```

Comment:
While comprehensions are the recommended solution, they are not the only one.

If I understand correctly, the concepts listed here should be *essential*, not *supplemental*. @ErikSchierboom , is that correct?

Reply:
@cmccandless Some languages do list different approaches, but that is not in any way requisite. I'd say: if you want to list more, do so, but the focus should be on the essential Concepts.

Reply:
Agree that comprehensions are never _required_ - then again, they are considered _pythonic_  by many and are widely used.  They're also a syntax that many struggle at first to create and read.   Because of that, I'd put them on a core concept list.  No - not essential,  but needed to feel really comfortable.

Reply:
I tend to think on the lines of @BethanyG here; we should probably strive to have the recommended solution be the most idiomatic one -- and these days comprehensions _are_ the most idiomatic way of handling the building of a single collection via a loop -- the concept files themselves can describe less idiomatic fallbacks.

Reply:
Are comprehensions _essential_ to Python? Well, no ... but then neither is any form of string formatting beyond concatenation, and really _all_ of the math can be done with bitwise operations ... like string formatting and the normal arithmetic operators though they are definitely an essential thing to know about and understand to write idiomatic Python.

I guess what I'm saying is that since there will never exist an exercise for which a comprehension is _necessary_, should we then not include them as a concept anywhere?